src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs

```diff
@@ -56,7 +56,7 @@ public CompiledIdentityConstraint(XmlSchemaIdentityConstraint constraint, XmlNam
             catch (XmlSchemaException e)
             {
                 e.SetSource(constraint.Selector);
-                throw e;
+                throw;
```

Comment:
Ah -- totally missed that. :)

I have to admit that this behavior is slightly surprising :) but I think it is the right behavior for throw.

LGTM!